### DISCUSSION TITLE
iOS version

### DISCUSSION THREAD

**Discussion by JoBergeron** [diasurgical/DevilutionX#2339] [General]
I see that there's an iOS version of devolution x in the mod section of the documentation. Why wasn't it merged back upstream? Is there a reason for it?

**Comment by AJenbo** [comment]
It was never fully finished, since it was done in a way that isn't generally maintainable (using cmake as the build system).

But we also lack a maintainer and the know how regarding how to distribute it in a way that lets people install it.

**Reply by StephenCWills** [reply]
To clarify, I believe AJenbo means to say that using cmake would be the generally maintainable way, and it doesn't do it that way.

**Reply by JoBergeron** [reply]
I'll try and take a look at it but it'll be a learning experience.

**Reply by AJenbo** [reply]
That would be much appropriated, and good timing considering that we are working on general phone support atm.

**Comment by Haken00** [comment]
Hi,
I have just recently found out about this project - as a long time Diablo fan it's truly amazing news for me.

Can I ask if there are plans to implement iOS version? Currently I believe it's available on Google play store - so I'm hoping porting it to AppStore will be happening as well? Needless to say this is one of the projects I'd be willing to pay for - and certainly many other iPhone users as well.

Also I've read that the Android version supports controllers - if there could be support for MFi (Razer Kishi and others) controllers on iOS that would be amazing.

Thanks
Jan

**Reply by qndel** [reply]
1. we can't charge money for it
2. don't think appstore would ever let us in

**Reply by AJenbo** [reply]
Regarding the MFi controller, the bug with them is handled here: https://github.com/diasurgical/devilutionX/pull/1294

We have iOS builds, but no way of sharing them. Users will probably have to build it themselves. So it would require a Mac to install. If you know of any solution to that then let us know as that is one of the main blockers.

**Reply by julealgon** [reply]
> don't think appstore would ever let us in

Out of curiosity, why is that the case @qndel ? Is there any specific requirement we don't meet?

**Reply by qndel** [reply]
> > don't think appstore would ever let us in
> 
> Out of curiosity, why is that the case @qndel ? Is there any specific requirement we don't meet?

dunno, considering we are doing this for free, even the 100$/year apple developer membership which seems to be required for publishing stuff on appstore is a giant nope, doing this crap for free is enough of a joke, paying for it? LOL
+dunno what's their attitude towards reverse engineered stuff, I just dont like apple

**Reply by JoBergeron** [reply]
Never used myself but there's something called the Altstore that sort of streamlines installing apps outside of the appstore. No jailbreaking required.

**Reply by AJenbo** [reply]
Thanks, that might actually work.

This looks relevant for porting:
    https://github.com/i8degrees/hello-sdl2-ios-cmake 166
    http://pastebin.com/yADa0R7f 35

**Reply by Haken00** [reply]
Hi,
if money is the major issue (suspect it isn't, at least not the only one), then surely the community would be willing to pay for your work on a platform such as Kickstarter? I suspect there would be other (legal among others) issues but if you managed to publish this in Google Play, I'm assuming you probably went through some sort of publishing process as well (that being said I'm not a developer and don't use android so my knowledge here is very very basic).

**Reply by JoBergeron** [reply]
Taking any money except for donations would be an invitation to get destroyed by Activision Blizzard lawyers I believe

**Reply by AJenbo** [reply]
Being on Google Play was not our original intention. Someone uploaded an app to Google Play in our name, that we weren't happy with the quality of. To resolve it we had to make an account ourselves.

**Comment by AJenbo** [comment]
Things are now mostly running but I need help with testing on a real device and some minor issues, please see this PR: #3612

**Comment by eduo** [comment]
I want to thank the team for a solid iOS/ipadOS (I haven't tried on appletv nor imagine it's yet a target).

Quick question: Are there docs for iOS/iPad? Buttons and joystick (for onscreen playing) are layered on top of UI and there's no obvious way to move or resize them. Is this an open item for the future or is there access to DevilutionX's settings separate from Diablo settings where this can be addressed? I may have just missed it if so.

**Reply by AJenbo** [reply]
Thanks. We currently do not expose a way to adjust the size of the touch controls. You will find most settings in the main menu (not options in-game).

**Reply by AntonKosenkoPro** [reply]
Is there still no way to move touch controls? iOS port is great, but I died so many times, when tapped slightly left or right around attack button and opened a spell book. I would move touch controls above the original UI.
![image](https://user-images.githubusercontent.com/2052851/210128925-a1b5cb45-5f4a-4c1c-b104-9cd9c126c142.png)

**Reply by AntonKosenkoPro** [reply]
So, I think I should research before ask. Finally, I found a workaround to this issue. You can change resolution and controls stop overlapping UI. But from settings it's possible to choose only between default and native (2160x1620 in my case), and in native resolution UI was too small. So I found that I can set custom resolution in diablo.ini file and by looking over several resolutions, I found that 960x720 is the best variant at my iPad.
![image](https://user-images.githubusercontent.com/2052851/210130378-34dea844-c3c6-45ab-b818-52bf30c45f4b.png)
